### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by Luckyaremu**
#tic tac toe miles stone two

-User interface set up
Thank you for reviewing are project if you notice any corrections needed, kindly let us know.

kind regards

### CODE REVIEW

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @Luckyaremu, @kotoku-code. Good work on this milestone:clap:. Let's now take care of the following :fast_forward:**
- The game is for two human players and not against the computer.
image

- Player should be informed whenever an invalid choice is made, say the move has already been made before or the move is not a valid one at all.

- The gameboard should be displayed after each turn.

- Gameplay should be represented by a loop that repeats game actions (ask for input, check for validity, check winning conditions, etc...) until there is a draw or loop.

- Remove the test.rb file from the project directory. It doesn't serve any purpose for the project now.

- **Optional:** A better way to check for winning conditions is to maintain a single array for the board positions and define a method that checks if the winning positions on the board are being occupied by the same player symbol, instead of hard-coding the winning positions.


_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by Luckyaremu** [comment]
#tic tac toe miles stone two

-User interface remodeled to accommodate two players
Thank you for reviewing are project if you notice any corrections needed, kindly let us know.

kind regards

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @Luckyaremu and @kotoku-code :balloon: :lollipop: 

### Thank you for doing the 2 player change and request for both players names, I think that adds value to your project and makes it more user friendly :video_game: 

### Sadly I couldn't play the game pass inputting the players' names because it retrieves an error

#### uninitialized constant Player (NameError)

- [ ] Fix this error and make sure everything requested by @mosesogwo works fine before submitting for a code review. 

image

### That's it, team! :raised_hands: 

### Please :pray: make this change :man_technologist: :man_technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Comment by kotoku-code** [comment]
made corrections from given feedback

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Nice Progress guys :+1: 
You have made tremendous progress in this project. However, we need to fix a few major features before merging.
- [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
- [ ] Inside the loop, all actions should be repeated for each move
- [ ] the game should be played between humans not between humans a computer
#### Check more of my comments for suggestions.

### Martin Afani

File: bin/main.rb [hunk 1]
```diff
@@ -1 +1,66 @@
-puts name = 'Hello World!'
+# !/usr/bin/env ruby
+require_relative '../lib/game_logic.rb'
+
+winningcondtionshuman = [1, 2, 3] || [4, 5, 6] || [7, 8, 9] || [1, 4, 7] ||
+                        [2, 5, 8] || [3, 6, 9] || [1, 5, 9] || [3, 5, 7]
+outputhuman = []
+outputcomputer = []
```
Comment (whiz25) [inline bin/main.rb]: - [ ] game should be played with humans

**Comment by kotoku-code** [comment]
corrections made from feedback

**Comment by Anna-Myzukina** [comment]
## Hi team ! Very nice project! :wave: 

# You did a great job, approved  :+1:

### Go and merge to develop.
`Goodluck in your next milestone Happy coding ` <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Review by Anna-Myzukina — APPROVED** [review]

**Pull request merged**